### PR TITLE
Aave and Spark on Omni

### PR DESCRIPTION
**Pull request by piekczyk**
# [Aave and Spark on Omni](https://www.notion.so/oazo/Moving-Aave-Spark-to-Omni-kit-e8f0442db1ce4491b4f009faa9ec9fdf)

<please insert a shortcut link above>
  
## Changes 👷‍♀️
  <Please add short list of changes>

- rewritten aave and spark to use omni
- general updates in the scope of omni
  
## How to test 🧪
  <Please explain how to test your changes>

- reach out to me

### CODE REVIEW

**Review by jakubswierczek — DISMISSED** [review]

**Review by jakubswierczek — APPROVED** [review]

File: features/aave/hooks/useAaveEarnYields.ts [hunk 1]
```diff
@@ -18,15 +18,23 @@ export function useAaveEarnYields(
...
-    aaveEarnYieldsQuery(riskRatio, yieldFields)
-      .then((yieldsResponse) => {
-        setYields(yieldsResponse)
-      })
-      .catch((e) => {
-        console.error('unable to get yields', e)
-      })
-  }, [aaveEarnYieldsQuery, riskRatio, yieldFields])
+    // Timeout added to debounce user input
+    setYields(undefined)
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/aave/hooks/useAaveEarnYields.ts]: fixed

File: features/omni-kit/components/details-section/OmniOpenYieldLoopFooter.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    environment: { quoteToken },
+  } = useOmniGeneralContext()
+  const {
+    position: { isSimulationLoading },
+    form: {
+      state: { depositAmount },
+    },
+  } = useOmniProductContext(OmniProductType.Multiply)
+
+  const amount = useMemo(() => depositAmount || new BigNumber(100), [depositAmount])
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/omni-kit/components/details-section/OmniOpenYieldLoopFooter.tsx]: fixed

File: features/omni-kit/components/details-section/OmniContentCard.tsx [hunk 3]
```diff
@@ -22,18 +26,22 @@ export interface OmniContentCardBase {
...
   unit?: string
   value?: OmniContentCardValue
+  link?: DetailsSectionContentCardLinkProps
 }
 
 export interface OmniContentCardExtra {
   asFooter?: boolean
   changeVariant?: ChangeVariantType
   customValueColor?: string
+  customBackground?: string
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/omni-kit/components/details-section/OmniContentCard.tsx]: removed

File: features/omni-kit/components/details-section/parsers/useOmniCardDataLiquidationRatio.ts
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/omni-kit/components/details-section/parsers/useOmniCardDataLiquidationRatio.ts:1]: removed

File: features/aave/services/get-aave-history-events.ts [hunk 4]
```diff
@@ -17,20 +18,31 @@ const sumStringNumbersArray = (numbersArray: { amount: string }[]): BigNumber =>
...
+  quoteToken: string,
 ): Promise<{
-  events: AaveHistoryEvent[]
-  positionCumulatives?: AaveCumulativeData
+  events: AaveLikeHistoryEvent[]
+  positionCumulatives?: AaveLikeCumulativeData
 }> {
-  const resposne = await loadSubgraph('Aave', 'getAaveHistory', _networkId, {
+  const tokens = getNetworkContracts(
+    _networkId as
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/aave/services/get-aave-history-events.ts]: fixed

File: features/omni-kit/contexts/OmniGeneralContext.tsx [hunk 5]
```diff
@@ -184,6 +188,35 @@ export function OmniGeneralContextProvider({
     }
   }
 
+  const getEntryToken = () => {
```
[the remark this answers is not in the document]
Comment (piekczyk) [reply features/omni-kit/contexts/OmniGeneralContext.tsx]: moved to separate file

**Review by piotrkonowrocki — APPROVED** [review]

**Pull request merged**